contracts/truefi2/distributors/BorrowerDistributor.sol

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.6.10;
+
+import {SafeMath} from "@openzeppelin/contracts/math/SafeMath.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import {UpgradeableClaimable} from "../../common/UpgradeableClaimable.sol";
+
+/**
+ * @title BorrowerRewardDistributor
+ * @dev Reward borrowers for taking out loans in TRU
+ */
+contract BorrowerRewardDistributor is UpgradeableClaimable {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+
+    // ================ WARNING ==================
+    // ===== THIS CONTRACT IS INITIALIZABLE ======
+    // === STORAGE VARIABLES ARE DECLARED BELOW ==
+    // REMOVAL OR REORDER OF VARIABLES WILL RESULT
+    // ========= IN STORAGE CORRUPTION ===========
+
+    /**
+     * @notice Will be TRU
+     */
+    IERC20 public rewardCurrency;
+
+    /**
+     * @notice Only lenders can adjust the reward rate and distribute the reward
+     */
+    address public lender;
+
+    /**
+     * @dev Reward rate is set as 1% by default. 2 decimals
+     * so the actual reward can be calculated as: loan * rewardRate / 10000
+     */
+    uint256 public rewardRate = 100;
+
+    modifier onlyLender() {
+        // Modifier
+        require(msg.sender == lender, "Only lender can call this.");
+        _;
+    }
+
+    /** @notice setting TRU address
+     */
+    function initialize(IERC20 _rewardCurrency, address _lender) external initializer {
+        rewardCurrency = _rewardCurrency;
+        lender = _lender;
+    }
+
+    /** @notice setting reward rate
+     */
+    function setRewardRate(uint256 _rewardRate) external onlyLender {
+        rewardRate = _rewardRate;
+    }
+
+    /** @notice Distribute reward
+     */
+    function distribute(address borrower, uint256 loanAmount) external onlyLender {
```

Comment:
We also need the loan decimals, otherwise we aren't properly scaling rewards for different loan currencies. I would pass in an ILoanToken2 and set `rewardAmount = loan.amount() * rewardCurrency.decimals() / loan.decimals() * rewardRate / BASIS_RATIO` (translated to SafeMath, of course). Maybe split this calculation into two lines for legibility?

You'll need to import `IERC20WithDecimals` from `contracts/truefi2/interface/' instead of OpenZeppelin's IERC20.

Reply:
Also if we're passing in a LoanToken2 then that should already include the borrower address `loan.borrower()`. So no need to pass it in as a separate arg.